addons/udes_stock/models/stock_picking.py

```diff
@@ -456,6 +456,16 @@ def assert_not_lot_restricted(self):
                             )
                         )
 
+    def button_draft(self):
+        """
+        Button that passes the picking to draft state. Adding extra check on backend that should be
+        triggered only for one transfer at time and state is on waiting or confirmed
+        when button is clicked.
+        """
+        self.ensure_one()
+        if self.state in ("waiting", "confirmed"):
+            self.write({"state": "draft"})
```

Comment:
Probably a silly question, but does this also unreserve the stock?

Reply:
Ah I see, so if stock has been reserved then the picking won't get put back to draft. The user only sees the button once they've unassigned stock.

Reply:
Very good question i think, i am assuming that shouldnt be any stock reserved when the button is clicked. Otherwise we shouldnt click the button set to draft.
Or maybe just in case to add an extra check if move lines exist to pass